search/node-search.go

```diff
@@ -26,8 +28,19 @@ type keyValue struct {
 	value *yaml.Node
 }
 
+// Parses filePath and attempts to create Node Object
+func MakeNode(filePath string) (*yaml.Node, error) {
+	var node yaml.Node
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	marshErr := yaml.Unmarshal(data, &node)
+	return &node, marshErr
+}
+
 // Given a path in a yaml file return the node at the end of the path
-func findNode(node *yaml.Node, path ...string) (*yaml.Node, error) {
+func FindNode(node *yaml.Node, path ...string) (*yaml.Node, error) {
```

Comment:
There are a few error conditions in this function that we should probably test.

Reply:
now testing to make sure that findKey( before FindNode) returns node with matching string as last item in path and that if the last item in path is an index, the returned node is a sequence and the index is in the bounds of the sequence

Reply:
The more detailed success test is awesome.  Verifying all output for a test is a good practice.  But, there are still a number of error states that should likely be tested.